app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
+
+  def saml_response_url
+    saml_request.response_url
+  end
```

Comment:
It was defined in the 18F fork of the gem but is not in the upstream gem. That's why I've added it here, since it seemed required specifically by our application.